vmdb/app/helpers/api_helper/renderer.rb

```diff
@@ -326,9 +326,16 @@ def render_resource_attr(resource, attr)
 
     def physical_attribute_selection(resource)
       return [] unless params['attributes']
-      physical_attributes = params['attributes'].split(",").collect do |attr|
-        attr unless attr_virtual?(resource, attr)
-      end.compact
+
+      @pas ||= {}
+      klass_name = resource.class.name
+      return @pas[klass_name] unless @pas[klass_name].nil?
+
+      @pas[klass_name] = gen_physical_attribute_selection(resource)
+    end
+
+    def gen_physical_attribute_selection(resource)
+      physical_attributes = params['attributes'].split(",").select { |attr| !attr_virtual?(resource, attr) }
```

Comment:
what you are calling physical attrbutes are really just columns, right?

If that's the case, you don't need the cache...can you just lookup with Model.columns_hash (which is cached already)?